src/Confluent.SchemaRegistry.Serdes.Json/JsonSerializer.cs

```diff
@@ -171,26 +173,35 @@ public async Task<byte[]> SerializeAsync(T value, SerializationContext context)
 
                     if (!subjectsRegistered.Contains(subject))
                     {
-                        if (useLatestVersion)
+                        if (autoRegisterSchema) 
+                        {
+                            schemaId = await schemaRegistryClient.RegisterSchemaAsync(subject,
+                                        new Schema(this.schemaText, EmptyReferencesList, SchemaType.Json), normalizeSchemas)
+                                    .ConfigureAwait(continueOnCapturedContext: false);
+                        } 
+                        else if(useLatestVersion)
                         {
                             var latestSchema = await schemaRegistryClient.GetLatestSchemaAsync(subject)
                                 .ConfigureAwait(continueOnCapturedContext: false);
+                            // checks if locally generated schema is compatibile with latest schema on schema and autoregister if set to true else throws exception. 
```

Comment:
this comment is not correct (nothing here auto registers the schema).

it's also unnecessary - doesn't say anything a quick look at the code will tell you, so remove it.

Reply:
remove this comment